lib/puppet/provider/file/windows.rb

```diff
@@ -71,7 +71,9 @@ def mode
 
   def mode=(value)
     begin
-      set_mode(value.to_i(8), resource[:path])
+      managing_group = resource[:group].nil? ? false : true
+      managing_owner = resource[:owner].nil? ? false : true
```

Comment:
nit: `.nil?` already returns a boolean so no need to use a ternary here.

`managing_owner = !resource[:owner].nil?`